### ISSUE TITLE
Externalise mdast plugin

### ISSUE THREAD

**Issue by wooorm** [documentationjs/documentation-readme#3]
Hey!

First of all, very nice work! I’m also happy that you actually made an mdast plugin instead of sealing pluggability off (like some people do :astonished:)

Second, I’d love to be able to use this as a plug-in through mdast’s (CLI) plug-in system without having to use a long path (`documentation-readme/lib/plugin`). I personally recommend externalising to `mdast-documentation` in lieu with other [mdast plugins](https://github.com/wooorm/mdast/blob/master/doc/plugins.md#list-of-plugins).

On a side node, I agree with @tmcw [suggestion](https://github.com/documentationjs/documentation-readme/commit/713986363c268d18bca2c295d8299f64b126e6a2#commitcomment-13492129), adding support for passing an object [to the plugin](https://github.com/documentationjs/documentation-readme/commit/713986363c268d18bca2c295d8299f64b126e6a2#diff-f85c36a1cc8952fb5e6fc7200a89ceaaR7) would be awesome. That would mean I won’t have to pass a weird command-line string [in `.mdastrc`](https://github.com/wooorm/mdast/blob/b28a45ac3d8291ef2fb3e173468ac992f8c56f1f/.mdastrc) files.

P.S. Ping me if you have Qs regarding mdast, I’d love to help :smile:

**Comment by anandthakker** [comment]
@wooorm Thanks!  Yes, I definitely feels like there's an `mdast` plugin worth separating out here, and am planning to do so soon...  I thought I'd give it a couple days to let the use case shake out a bit first.

I'm thinking that instead of `mdast-documentation`, maybe it should be generalized to something like `mdast-inject`, which could take an options object like `{ section: "headingToInjectContent", content: string/md-returning-function/stream }`.  Seems like that would be more widely useful than a `documentation`-specific one...

**Comment by wooorm** [comment]
@anandthakker Oh okay, great to hear!

I made a choice separating “plug-ins” which can be used directly by a user (also on the CLI) into what I call [plug-ins](https://github.com/wooorm/mdast/blob/master/doc/plugins.md#list-of-plugins) (prefixed with `mdast-`), and utilities which help with AST transformations etc. into [utilities](https://github.com/wooorm/mdast/blob/master/doc/plugins.md#list-of-utilities) (prefixed with `mdast-util-`). The latter is often used by other plugins, which is why I’m telling you this: your example of injecting a stream into a “section” seems really useful to other developers but less so for people which just plan on use a couple plug-ins (and his naming convention helps them not accidentally getting the wrong thing).

Long story short: AWESOME, and I’d suggest using `mdast-util-inject` as a name for the injection functionally, and sticking to `mdast-documentation` for the `documentationjs` integration!

**Comment by anandthakker** [comment]
Ah, cool.  Your plugin/utility distinction makes sense to me.

**Comment by wooorm** [comment]
:+1:

**Comment by tmcw** [comment]
I'm going to do a minor rearrange in this direction

**Comment by anandthakker** [comment]
Decided to externalize the utility (https://github.com/anandthakker/mdast-util-inject), but not the plugin for now, basically because it feels like the dependency structure makes more sense that way (see https://github.com/documentationjs/documentation/pull/313)
